internal/repository/postgresdb/appreciation.go

```diff
@@ -0,0 +1,320 @@
+		LeftJoin(appr.TableRewards+" r ON a.id = r.appreciation_id").
+		Where(squirrel.And{
+			squirrel.Eq{"a.id": apprId},
+			squirrel.Eq{"a.is_valid": true},
+		}).
+		GroupBy("a.id","cv.name","u_sender.id","u_receiver.id").
+		ToSql()
+
+	if err != nil {
+		logger.Errorf("error in generating squirrel query, err: %v", err)
+		return repository.AppreciationInfo{}, apperrors.InternalServer
+	}
+
+	var resAppr repository.AppreciationInfo
+
+	// Execute the query
+	err = queryExecutor.QueryRowx(query, args...).StructScan(&resAppr)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			logger.Warn(fmt.Sprintf("no appreciation found with id: %d", apprId))
+			return repository.AppreciationInfo{}, apperrors.AppreciationNotFound
+		}
+		logger.Errorf("failed to execute query: %v", err)
+		return repository.AppreciationInfo{}, apperrors.InternalServer
+	}
+	return resAppr, nil
+}
+func (appr *appreciationsStore) ListAppreciations(ctx context.Context, tx repository.Transaction, filter dto.AppreciationFilter) ([]repository.AppreciationInfo, repository.Pagination, error) {
+
+	queryExecutor := appr.InitiateQueryExecutor(tx)
+
+	// Get logged-in user ID
+	data := ctx.Value(constants.UserId)
+	userID, ok := data.(int64)
+	if !ok {
+		logger.Error("err in parsing userID from token")
+		return []repository.AppreciationInfo{}, repository.Pagination{}, apperrors.InternalServerError
+	}
+
+	// query builder for counting total records
+	countQueryBuilder := repository.Sq.Select("COUNT(*)").
+		From("appreciations a").
+		LeftJoin("users u_sender ON a.sender = u_sender.id").
+		LeftJoin("users u_receiver ON a.receiver = u_receiver.id").
+		LeftJoin("core_values cv ON a.core_value_id = cv.id").
+		Where(squirrel.Eq{"a.is_valid": true})
+	if filter.Name != "" {
+		lowerNameFilter := fmt.Sprintf("%%%s%%", strings.ToLower(filter.Name))
+		countQueryBuilder = countQueryBuilder.Where(
+			"(LOWER(CONCAT(u_sender.first_name, ' ', u_sender.last_name)) LIKE ? OR "+
+				"LOWER(CONCAT(u_receiver.first_name, ' ', u_receiver.last_name)) LIKE ?)",
+			lowerNameFilter, lowerNameFilter,
+		)
+	}
+	
+	if filter.Self {
+		countQueryBuilder = countQueryBuilder.Where(squirrel.Or{
+			squirrel.Eq{"a.sender": userID},
+			squirrel.Eq{"a.receiver": userID},
+		})
+	}
+
+	countSql, countArgs, err := countQueryBuilder.ToSql()
+	if err != nil {
+		logger.Error("failed to build count query: ", err.Error())
+		return []repository.AppreciationInfo{}, repository.Pagination{}, apperrors.InternalServerError
+	}
+
+	var totalRecords int32
+	err = queryExecutor.QueryRowx(countSql, countArgs...).Scan(&totalRecords)
+	if err != nil {
+		logger.Error("failed to execute count query: ", err.Error())
+		return []repository.AppreciationInfo{}, repository.Pagination{}, apperrors.InternalServerError
+	}
+
+	pagination := getPaginationMetaData(filter.Page, filter.Limit, totalRecords)
+	// Initialize the Squirrel query builder
+	queryBuilder := repository.Sq.Select(
```

Comment:
no need to create a new query again, you can use existing queryBuilder and update the Select part only, check this in squirrel documentation once. https://github.com/Masterminds/squirrel/blob/v1.5.4/select.go#L268

Reply:
updated